torch/onnx/_internal/fx/passes/type_promotion.py

```diff
@@ -1705,10 +1699,8 @@ def _run(self, *args, **kwargs) -> torch.fx.GraphModule:
         fake_args = self._fetch_fake_args()
         fake_mode = self.fake_mode
         assert fake_mode is not None, "Cannot detect fake_mode."
-
-        with fake_tensor.unset_fake_temporarily(), (
-            fake_mode
-        ), fx_traceback.preserve_node_meta():
+        dispatcher_mode = torch._dispatch.python.enable_python_dispatcher()
```

Comment:
How about put a NOTE to remind the importance of `enable_python_dispatcher`, and maybe there is no test to guard this case, and then link to the issue or this PR.

Reply:
Done, thanks!